core/src/com/unciv/logic/battle/Battle.kt

```diff
@@ -362,16 +368,34 @@ object Battle {
     ) {
         // implementation based on the description of the original civilopedia, see issue #4374
         if (plunderingUnit !is MapUnitCombatant) return
+        val civ = plunderingUnit.getCivInfo()
         val plunderedGoods = Stats()
 
         for (unique in plunderingUnit.unit.getMatchingUniques(UniqueType.DamageUnitsPlunder, checkCivInfoUniques = true)) {
             if (plunderedUnit.matchesFilter(unique.params[1])) {
```

Comment:
Since the body has gotten big with its own nesting, please reverse this if (if !matches() continue)